Lib/pdb.py

```diff
@@ -1987,6 +2004,10 @@ def _run(self, target: Union[_ModuleTarget, _ScriptTarget]):
         __main__.__dict__.clear()
         __main__.__dict__.update(target.namespace)
 
+        # Clear the mtime table for program reruns, assume all the files
+        # are up to date.
+        self._file_mtime_table.clear()
```

Comment:
Can we cover this with a test?

Reply:
A test is added for this - confirmed it fails without clearing the table.